core/src/mender-client.c

```diff
@@ -1050,6 +1055,56 @@ mender_prepare_new_provides(mender_artifact_ctx_t *mender_artifact_ctx, char **n
 
     return MENDER_OK;
 }
+
+static mender_err_t
+mender_check_device_compatibility(mender_artifact_ctx_t *mender_artifact_ctx) {
+
+    /* We need to load the stored provides */
+    mender_key_value_list_t *stored_provides = NULL;
+    if (MENDER_FAIL == mender_storage_get_provides(&stored_provides)) {
```

Comment:
```suggestion
    if (MENDER_OK != mender_storage_get_provides(&stored_provides)) {
```

Reply:
`mender_storage_get_provides` can return `MENDER_NOT_FOUND`. With the current design we still need to check if there are unmet depends if there are no provides in the storage.

Reply:
Ok, it makes sense.

Alternatively we could have an early check for the Artifact having dependencies at all. If there are none we can skip the check (return ok) and if there are then here we must have provides in the store (or return error). This is just thinking out loud, I think the current code is good as-is